sql-tuning-best-practice.md

```diff
@@ -0,0 +1,952 @@
+
+```sql
+SHOW VARIABLES LIKE 'tidb\_auto\_analyze%';
+```
+
+```
++-----------------------------------------+-------------+
+| Variable_name                           | Value       |
++-----------------------------------------+-------------+
+| tidb_auto_analyze_ratio                 | 0.5         |
+| tidb_auto_analyze_start_time            | 00:00 +0000 |
+| tidb_auto_analyze_end_time              | 23:59 +0000 |
++-----------------------------------------+-------------+
+```
+
+There are cases where automatic collection doesn't meet your needs. The analyze windown by default is 00:00 to 23:59, which means the analyze job can be triggered any time during the day. If you want to trigger the analyze job only during certain hours, you can set the start time and end time, to avoid performance impact for the online business.
+
+You can manually collect statistics using the `ANALYZE TABLE table_name` statement. This allows you to:
+
+1. Adjust the sample rate for more accurate or faster analysis
+2. Increase the number of top-N values collected
+3. Gather statistics for specific columns only or for all columns
+
+It's important to note that after manual collection, subsequent automatic gathering jobs will inherit the new parameters. This means that any customizations you've made during manual collection will be carried forward in future automatic analyses.
+
+Another common scenario is locking table statistics. This is useful when:
+
+1. The statistics on the table are already representative of the data.
+2. The table is very large and statistics collection is time-consuming.
+3. You want to maintain statistics only during specific time windows.
+
+To lock the statistics for a table, you can use the following statement [`LOCK STATS table_name`](/sql-statements/sql-statement-lock-stats.md).
+
+for more detail about statistics, please refer to [statistics](/statistics.md).
+
+## How TiDB Builds A Execution Plan
+
+An SQL statement undergoes optimization primarily in the optimizer through three stages:
+
+- Pre-Processing
+- Logical Transformation
+- Cost-based Optimization
+
+### Pre-Processing
+
+The main actions in the pre-processing stage it to determine if the SQL statement can be executed by using [`Point_Get`](https://docs.pingcap.com/tidb/stable/explain-indexes#point_get-and-batch_point_get) or [`Batch_Point_Get`](https://docs.pingcap.com/tidb/stable/explain-indexes#point_get-and-batch_point_get).
+
+`Point_Get` or `Batch_Point_Get` is to get 1 or 0 or many row only by using the TiKV key, the explicit or implicit (`_tidb_rowid`) primary key. For example, when id column is the primary key of a [clustered index](/clustered-indexes.md) table, `Point_Get` is used to get the particular row. If a plan is identified as `Point_Get`, optimizer will skip the logical transformation and cost-based optimization.
```

Comment:
Or maybe:
```suggestion
`Point_Get` or `Batch_Point_Get` means using a primary or unique key, to directly read from TiKV, by exact key lookup. For example, when id column is the primary key of a [clustered index](/clustered-indexes.md) table, `Point_Get` is used to get the particular row. If a plan is identified as `Point_Get`, optimizer will skip the logical transformation and cost-based optimization, since the exact key read will be the best way to access the row. `Batch_Point_Get` will read a batch of exact key lookups, like `SELECT * FROM t WHERE id IN (4,62,81)`

Example:
```